src/Storages/StorageFile.cpp

```diff
@@ -581,32 +592,54 @@ void StorageFile::truncate(
 
 void registerStorageFile(StorageFactory & factory)
 {
+    StorageFactory::StorageFeatures storage_features{
+        .supports_settings = true,
+        .source_access_type = AccessType::FILE
+    };
+
     factory.registerStorage(
         "File",
-        [](const StorageFactory::Arguments & args)
+        [](const StorageFactory::Arguments & factory_args)
         {
-            ASTs & engine_args = args.engine_args;
+            StorageFile::CommonArguments storage_args{
+                .table_id = factory_args.table_id,
+                .columns = factory_args.columns,
+                .constraints = factory_args.constraints,
+                .context = factory_args.context
+            };
 
-            if (!(engine_args.size() >= 1 && engine_args.size() <= 3)) // NOLINT
+            ASTs & engine_args_ast = factory_args.engine_args;
+
+            if (!(engine_args_ast.size() >= 1 && engine_args_ast.size() <= 3)) // NOLINT
                 throw Exception(
                     "Storage File requires from 1 to 3 arguments: name of used format, source and compression_method.",
                     ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);
 
-            engine_args[0] = evaluateConstantExpressionOrIdentifierAsLiteral(engine_args[0], args.local_context);
-            String format_name = engine_args[0]->as<ASTLiteral &>().value.safeGet<String>();
+            engine_args_ast[0] = evaluateConstantExpressionOrIdentifierAsLiteral(engine_args_ast[0], factory_args.local_context);
+            storage_args.format_name = engine_args_ast[0]->as<ASTLiteral &>().value.safeGet<String>();
 
-            String compression_method;
-            StorageFile::CommonArguments common_args{
-                args.table_id, format_name, compression_method, args.columns, args.constraints, args.context};
+            if (factory_args.storage_def->settings)
+            {
+                Context local_context_copy = factory_args.local_context;
+                local_context_copy.applySettingsChanges(
+                    factory_args.storage_def->settings->changes);
+                storage_args.format_settings = getFormatSettings(
+                    local_context_copy);
```

Comment:
But will it respect now the settings provided at query time?  Not of engine=File creation? 

I.e. 
```
create table x...
set input_format... 
select from x
```
If not - may lead to compatibility issues.

Reply:
It's going to ignore them, see https://github.com/ClickHouse/ClickHouse/pull/16591#issuecomment-722321307